File: app/CausalityCSharp/CausalityLibrary/Type/Scenario.cs
using System.Collections.Generic;

namespace CausalityLibrary.Type
{
    public sealed class Scenario
    {
        string SerialNumber
        { get; }

        public string Description
        { get; }

        public bool Displayed
        { get; set; }

        public readonly string[] Scenes;

        public Scenario(string serialNumber, string description,
             string[] scenes, bool displayed = false)
        {
            SerialNumber = serialNumber;
            Description = description;
            Scenes = scenes;
            Displayed = displayed;
        }
    }
}

File: app/CausalityJava/src/com/inha/dsp/causality/type/Scenario.java
package com.inha.dsp.causality.type;

import java.io.Serializable;

public class Scenario implements Serializable {
    private SerialNumber serialNumber;
    private String description;

    public SerialNumber getSerialNumber() {
        return serialNumber;
    }
    public String getDescription() {
        return description;
    }
    public boolean Displayed;
    public final SerialNumber[] Scenes;

    public Scenario(SerialNumber sn, String desc, SerialNumber[] scenes,
                    boolean displayed)
    {
        serialNumber = sn;
        description = desc;
        Scenes = scenes;
        Displayed = displayed;
    }

    // Don't instantiate with default constructor!
    Scenario() {
        Scenes = null;
    }
}
